docs/components/concepts/execution-listeners.md

```diff
@@ -0,0 +1,352 @@
+        </extensionElements>
+    </serviceTask>
+    <boundaryEvent attachedToRef="mainActivity" id="boundarySignalEvent" name="boundary_signal_event">
+        <extensionElements>
+            <zeebe:executionListeners>
+                <zeebe:executionListener eventType="end" retries="3" type="end_execution_listener_job"/>
+            </zeebe:executionListeners>
+        </extensionElements>
+        <signalEventDefinition id="signalEventDefinition" signalRef="signalRef"/>
+    </boundaryEvent>
+    <!-- other elements -->
+</process>
+<signal id="signalRef" name="my_signal"/>
+```
+
+</details>
+
+### Gateways (End ELs)
+
+**Workaround:** Use `start` ELs on the element following the gateway to execute the required logic. This allows handling of any post-execution tasks in a dedicated element.
+
+<details>
+  <summary><strong>Example:</strong></summary>
+
+```xml
+<!-- other elements -->
+<exclusiveGateway default="toServiceTaskFlow" id="exclusiveGateway" name="exclusiveGateway">
+    <outgoing>toServiceTaskFlow</outgoing>
+</exclusiveGateway>
+<sequenceFlow id="toServiceTaskFlow" sourceRef="exclusiveGateway" targetRef="task"/>
+<serviceTask id="task" name="task">
+    <extensionElements>
+        <zeebe:taskDefinition type="service_task_after_gateway"/>
+        <zeebe:executionListeners>
+            // highlight-next-line
+            <zeebe:executionListener eventType="start" type="task_after_gateway_start_listener"/>
+        </zeebe:executionListeners>
+    </extensionElements>
+    <incoming>toServiceTaskFlow</incoming>
+</serviceTask>
+<!-- other elements -->
+```
+
+</details>
+
+By understanding these workarounds, you can effectively design workflows to accommodate the limitations of ELs
+and maintain the flexibility and control of your process executions.
+
+## Usage
+
+### Job Workers
+
+To process EL jobs, a Job Worker should be defined. Job Workers handle specific tasks by subscribing to jobs of a
+particular type and executing the associated business logic. They allow for flexible and reusable logic in workflows.
+For more detailed information on how to create and manage Job Workers,
+refer to the [Camunda Job Workers documentation](/components/concepts/job-workers.md) & [Job Worker example](apis-tools/java-client-examples/job-worker-open.md).
+
+## Variables
+
+### Scope
+
+Variables set during the completion of an EL job are local to the element where the EL is defined.
+This design choice helps to prevent global variable scope cluttering, ensuring that the variables
+are contained within the context of the specific element they are associated with.
+
+### Dynamic variable handling
+
+Execution Listeners enable the dynamic fetching and calculation of variables for the next steps in the process,
+based on the current data. This functionality is particularly useful for elements such as gateways and events,
+allowing for more informed decision-making and precise control over process flow.
```

Comment:
:wrench: The usage of ELs and its technical base is most relevant for the user. We should move his part to the top.

:wrench: We should improve the wording to make it easier for users to understand.